aspect/make_variables.bzl

```diff
@@ -1,98 +1,194 @@
-"""Utility functions to expand make variables."""
-
-def _is_valid_make_var(varname):
-    """Check if the make variable name seems valid."""
-    if len(varname) == 0:
-        return False
-
-    # According to gnu make, any chars not whitespace, ':', '#', '=' are valid.
-    invalid_chars = ":#= \t\n\r"
-    for n in range(0, len(invalid_chars)):
-        if invalid_chars[n] in varname:
-            return False
-    return True
-
-def expand_make_variables(attr_name, expression, ctx, additional_subs = {}):
-    """Substitutes make variables defined in $() syntax.
-
-    Because ctx.expand_make_variables is deprecated, we need to be able to do the
-    substitution without relying on it.
-    Before the aspect is processed, the build system already detects most/all of
-    the failure modes and the aspect does not get processed, but including them
-    here helps with following the logic.
-
-    Args:
-      attr_name: The attribute name. Used for error reporting.
-      expression: The expression to expand. It can contain references to "Make
-        variables".
-      ctx: The context containing default make variables to subtitute.
-      additional_subs: Additional substitutions to make beyond the default make
-        variables.
-
-    Returns:
-      Returns a string after expanding all references to "Make variables". The
-      variables must have the following format: $(VAR_NAME). Also, $$VAR_NAME
-      expands to $VAR_NAME.
-
-
-    """
-    if "$" not in expression:
-        return expression
-
-    current_offset = 0
-    rv = ""
-    substitutions = {}
-    substitutions.update(ctx.var)
-
-    # make variables from ctx.var can be overridden
-    substitutions.update(additional_subs)
-
-    # skylark does not support while. This is the maximum iteration count this
-    # loop will need, but it will exit early if possible.
-    for _n in range(0, len(expression)):
-        if current_offset >= len(expression):
+"""Utility functions to expand make variables. Implementation taken from cc_helper. """
+
+# Source: https://github.com/bazelbuild/bazel/blob/6f7faa659e5eb3e56c8a6274ebcb86884703d603/src/main/starlark/builtins_bzl/common/cc/cc_helper.bzl#L910
```

Comment:
license?

Reply:
The bazel implementation is Apache License, Version 2.0. Do we need to mention this here?

Reply:
yep the license should be preserved